### DISCUSSION TITLE
Using Incremental Static Regeneration - Do I have to manually open my webpages for SEO?

### DISCUSSION THREAD

**Discussion by gyinshen** [vercel/next.js#34376] [Help]
I have a Nextjs site with over 300,000 pages. very low traffic at the moment so I hope I can improve SEO. Only 700+ pages were created a build time so do I have to manually open the rest of the documents / pages so that the static pages can be created for search engines to crawl? I'm thinking of using some automated downloader / scraper to do this as I have all the web addresses.

I have also uploaded sitemaps but Google has not accepted/fetch them (I was told this could take weeks). I think if I have static pages ready, it could improve my site's discoverability.

**Comment by Thamasky** [comment]
Crawlers will be able to fetch your pages, even when they aren't generated yet.

Crawling the pages won't update the cache for the static pages, so there won't be any revalidation happening yet.
However, it will get the right content. I think it just bypasses the cache layer.

**Comment by stevemarksd** [comment]
You can use `await res.unstable_revalidate(fullPath);`.

I have a custom server and when the server starts, will call a next.js api, that calls `await res.unstable_revalidate(fullPath);`.

Your custom server can fetch all your pages, and for each page just call the api. 

You can revalidate the most popular pages first to deploy faster.
